### DISCUSSION TITLE
SSR issue on Swiper Element (WebComponent)

### DISCUSSION THREAD

**Discussion by amitsarkerr** [nolimits4web/swiper#7801] [General]
Hi,

I’m using Swiper Element (Web Component) in a Next.js 15.0.3 project and have encountered a few issues:
	1.	**SSR Issue**: When rendering the component, I face challenges with Server-Side Rendering. Specifically, during image loading, the layout takes time to adjust, or columns expand unexpectedly when using the carousel.
	2.	**TypeScript Error**: I’m also encountering TypeScript errors related to the Swiper Element Web Component.

Are these known issues, or am I possibly missing some configuration or setup?

Thank you in advance for your help!

**Comment by mahmoudAcm** [comment]
@amitsarkerr  You can check this for the first point [solution](https://github.com/nolimits4web/swiper/discussions/7340#discussioncomment-11227184)
